### PR TITLE
YAHOO-119 and YAHOO-120

### PR DESCRIPTION
**Pull request by aleksgor**

### CODE REVIEW

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorExtention.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+		Assert.assertEquals(parameters[12], "12345");
+		Assert.assertEquals(parameters[13], "localhost");
+		Assert.assertEquals(parameters[14], "8040");
+
+		tmpFile.delete();
+		File commandfile=new File(shFile);
+		commandfile.delete();
+	}
+	
+	 private String writeScriptLoclgizeFile() throws IOException {
```
Comment (dennisyv) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorExtention.java]: pls correct mistake in method name

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorExtention.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  }
+
+  @Test(expected = IOException.class)
+  public void testInitException() throws Exception {
+	  Configuration config= new Configuration();
+	  config.set("yarn.nodemanager.linux-container-executor.path", "/bin/sh");
+	  LinuxContainerExecutor executer= new LinuxContainerExecutor();
+	  executer.setConf(config);
+	  executer.init();
+	// should be born IOException !
```
Comment (dennisyv) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorExtention.java]: may be we need to check which exception exactly we receive ?
for example check that IOException's cause is ExitCodeException and it's exitCode equals to smth

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorExtention.java (hunk 1 above)
Comment (aleksgor) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorExtention.java]: what method do you mean ?

Gorshkov Aleksey
IBM Certified
programmer 
Grid Dynamics
 +7 952 203 9796
Skype:aleksgorpsk
www.griddynamics.com
email: agorshkov@griddynamics.com

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorExtention.java (hunk 2 above)
Comment (aleksgor) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorExtention.java]: The ExitCodeException dies in the class LinuxContainerExecutor. 
I see only IOException. 
I can check message like 
Assert.assertEquals(e.getMessage(), "Linux container executor not configured properly (error=2)"); 
only

Gorshkov Aleksey
IBM Certified
programmer 
Grid Dynamics
 +7 952 203 9796
Skype:aleksgorpsk
www.griddynamics.com
email: agorshkov@griddynamics.com

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorExtension.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+		Assert.assertEquals(parameters[12], "12345");
+		Assert.assertEquals(parameters[13], "localhost");
+		Assert.assertEquals(parameters[14], "8040");
+
+		tmpFile.delete();
+		File commandfile=new File(shFile);
+		commandfile.delete();
+	}
+	
+	 private String writeScriptLoclgizeFile() throws IOException {
```
Comment (dennisyv) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestLinuxContainerExecutorExtension.java]: mistake

**Pull request closed without merging**